server/plugin/command.go

```diff
@@ -615,6 +776,30 @@ func getAutocompleteData(config *Configuration) *model.AutocompleteData {
 
 	github.AddCommand(issue)
 
+	webhook := model.NewAutocompleteData("webhook", "[command]", "Available commands: add , list")
+	webhookList := model.NewAutocompleteData(list, "[owner/repo]", "List webhooks or an organization or repository.")
+	webhookList.AddTextArgument("Owner/repo to list webhooks from", "[owner/repo]", "")
+
+	webhook.AddCommand(webhookList)
+	webhookAdd := model.NewAutocompleteData(add, "[owner/repo] callback_url token content_type events insecure_ssl", "Add a webhook to desired owner[/repo] with optional --config")
+	webhookAdd.AddTextArgument("Owner/repo to create a webhook", "owner[/repo]", "")
+	webhookAdd.AddTextArgument("Call Back URL", "", "")
+	webhookAdd.AddTextArgument("token", "", "")
+	webhookAdd.AddTextArgument("content_type", "", "")
+	webhookAdd.AddTextArgument("Comma-delimited list of one or more of: create, delete, check_run, check_suite, code_scanning_alert, member, commit_comment, deploy_key, deployment_status, deployment, discussion_comment, discussion, fork, issue_comment, issues, label, meta, milestone, package, page_build, project_card, project_column, project, pull_request_review_comment, pull_request_review_thread, pull_request_review, pull_request, push, registry_package, release, repository, repository_import, repository_vulnerability_alert, secret_scanning_alert, star, status, team_add, public, watch, gollum . Defaults to *", "[features] (optional)", `/[^,-\s]+(,[^,-\s]+)*/`)
```

Comment:
I think the point of this feature is to make it so the admin doesn't need to think about the webhook. We don't want to simply "move" the burden to our platform. Here's how I think this feature should work:

The admin runs `/github subscriptions add ...` to create a subscription. We check to see if we have a webhook set up for this repo or the surrounding org. If one exists, we're good to go and nothing left to do except create our own internal subscription as usual.

Otherwise, we reply back to the user: "No webhook was found for this repository or organization. Would you like the webhook to be created?", and show a form in the ephemeral response using an [Interactive Message](https://docs.mattermost.com/developer/interactive-messages.html). If the user has privileges to create an org-level webhook, provide that as an option on the form.

When the form is submitted, we create the appropriate webhook and automatically choose to the webhook events that the plugins needs to use. We can include the URL to the webhook that we created in our final output.

We should then process the original command submission of `/github subscriptions add ...`, so that information needs to be stored in the form. You can use the `Context` field of [PostActionIntegration](https://pkg.go.dev/github.com/mattermost/mattermost-server/model#PostActionIntegration
) to store the information about the originally submitted command.

Once that is done, we respond to the user with a link to the GitHub webhook page, and a message saying the subscription was created.

@sibasankar-demansol @hanzei @aaronrothschild Thoughts on this?

Reply:
While an interactive message is a very handy way to create the webhook, I would also be fine simply making use of https://github.com/mattermost/mattermost-plugin-github/pull/456 and returning an ephemeral post that contains the command that needs to be run.

Reply:
@hanzei  currently the scope of oAuth is for Repo level , could you let me know should we add that to scope 

rest suggestion i will work on

Reply:
@hanzei I like the slash command idea the ephemeral message idea. 2/5 I think we should show 2 commands, one for making a repo-level webhook, and one for making an org-level webhook.

Reply:
@hanzei  , @mickmister  
i have few things from my side
1. currently #456  is yet to be done and is incomplete until we give the oAuth access for Org level or move on with creating webhook on Repo level .
2. Regarding this issue do you want me to show a ephemeral message with 2 commands for repo and org level differently , as the slash command is not yet completed to create webhook only way left is to give a link redirecting to github plugin console

Reply:
@sibasankarnayak 1/5 How about we assume the other PR is complete, and we put the two slash commands here. Then  we can put the "Do Not Merge/Awaiting PR" label on this PR to signal that it's waiting on changes.

Reply:
@mickmister  Thats sounds good,

just have a thought of show 
when user subscribe to a individual repo can show 2 commands
and when user subscribe to org then we can return one command to create org hook

what do you think does it makes any sense

Reply:
@mickmister  @hanzei 
can you suggest the changes over visual representation 
 ![Screenshot 2021-08-19 at 5 57 12 PM](https://user-images.githubusercontent.com/86650698/130111738-0b83548b-b7d5-43a4-9f28-0298b50cda02.png)

Reply:
The response should make it clear that the command needs to be run. Also there is a  `/` missing at the beginning.

Are both webhooks needed in any case? Is the org specific enough?

Reply:
have updated the message , can you check now

![Screenshot 2021-08-25 at 9 59 58 PM](https://user-images.githubusercontent.com/86650698/130829374-3f367ce2-3d8e-4b8a-8bde-5a2b0ea62dda.png)

Reply:
@sibasankarnayak Make sure you either mention or re-request review from the person you're communicating with on in issues and PRs. The other person doesn't get a push notification unless you do one or the other. The one time you don't need to mention someone is when you are commenting on their issue or PR. They will automatically get notifications for the comment.

When you have made changes that you need to be reviewed (which is the case here), your comment here is fine but you should also explicitly re-request review so the reviewer knows they need to review again.

Reply:
Actually since you wouldn't be requesting a full re-review here, you can just mention the reviewer in your comment cc @hanzei

Reply:
@mickmister  Sure will keep this in mind

Reply:
> The response should make it clear that the command needs to be run. Also there is a `/` missing at the beginning.
> 
> Are both webhooks needed in any case? Is the org specific enough? Can the text make the answer to these questions clear?

@sibasankarnayak Please make sure you address @hanzei's points here

Reply:
@sibasankarnayak We should make it clear that one command will create a webhook for the repo, and the other will create one for the whole org. And we should explain that if the webhook is made for the org, then you won't need to create another webhook for any repos in that org going forward.